libexec/charliecloud.py

```diff
@@ -0,0 +1,637 @@
+import argparse
+import collections
+import copy
+import http.client
+import json
+import logging
+import os
+import shutil
+import sys
+import tarfile
+import types
+
+import version
+
+
+## Imports not in standard library ##
+
+# These are messy because we need --version and --help even if a dependency is
+# missing.
+
+# List of dependency problems.
+depfails = []
+
+try:
```

Comment:
I don't think it's necessarily a problem yet, but at some point we're going to want to remove the things that only ch-grow needs from this file. Currently, it's a mix of ch-grow, ch-tug, and ch-run-oci stuff, and we'll probably need to add more stuff for future python scripts.

Reply:
I agree; we'll get it cleaned up over time.